### PR TITLE
Use settings for  lucene Analyzer and version

### PR DESCRIPTION
**Pull request by infofromca**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs [hunk 1]
```diff
@@ -16,22 +16,22 @@ namespace OrchardCore.Lucene
...
-            LuceneIndexingService luceneIndexingService,
+            LuceneSettingsService _luceneSettingsService,
             LuceneAnalyzerManager luceneAnalyzerManager,
             ILuceneQueryService queryService,
             ILiquidTemplateManager liquidTemplateManager,
             ISession session)
         {
             _luceneIndexProvider = luceneIndexProvider;
-            _luceneIndexingService = luceneIndexingService;
+            this._luceneSettingsService = _luceneSettingsService;
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs]: Why using `this.` here?

File: src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs [hunk 2]
```diff
@@ -16,22 +16,22 @@ namespace OrchardCore.Lucene
...
+        private readonly LuceneAnalyzerSettingsService _luceneSettingsService;
         private readonly LuceneAnalyzerManager _luceneAnalyzerManager;
         private readonly ILuceneQueryService _queryService;
         private readonly ILiquidTemplateManager _liquidTemplateManager;
         private readonly ISession _session;
 
         public LuceneQuerySource(
             LuceneIndexManager luceneIndexProvider,
-            LuceneIndexingService luceneIndexingService,
+            LuceneAnalyzerSettingsService _luceneSettingsService,
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs]: LuceneAnalyzerSettingsService luceneSettingsService

File: src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs (hunk 1 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneQuerySource.cs]: remove `this.`

**Comment by sebastienros** [comment]
Would it make sense to have an analyzer per index? But I assume we can always change it later.

**Comment by sebastienros** [comment]
Let me know if you don't want to change the setting to go per index, if so I would do it myself, i think this is important as we could have specific indexes for different cultures, or even use an index to store keywords. I assume we could also pass the analyzer to the methods instead of having each of them load the analyzer. So if we call a method which requires an index name, we would also pass an analyzer.

File: src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexManager.cs [hunk 3]
```diff
@@ -52,19 +54,19 @@ LuceneAnalyzerManager luceneAnalyzerManager
...
             _rootDirectory = Directory.CreateDirectory(_rootPath);
+            _luceneAnalyzerSettingsService = luceneAnalyzerSettingsService;
             _luceneAnalyzerManager = luceneAnalyzerManager;
         }
 
-        public void CreateIndex(string indexName)
+        public async Task CreateIndex(string indexName)
         {
-            Write(indexName, _ => { }, true);
+           await Write(indexName, _ => { }, true);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexManager.cs]: When a method doesn't run anything after an async call then you don't need to await it, just return the task. In this case the method will become:

```
public Task CreateIndex(string indexName)
{
  return Write(indexName, _ => { }, true);
}
```

**Comment by infofromca** [comment]
@sebastienros : just changed it as your suggestion.
For others, I will leave it to you

File: src/OrchardCore.Modules/OrchardCore.Lucene/Views/LuceneAnalyzerSettings.Edit.cshtml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+@using Lucene.Net.Util
+@model LuceneSettingsViewModel
+<fieldset class="form-group" asp-validation-class-for="Version">
+    <label asp-for="Version" class="control-label"><span asp-validation-for="Version"></span></label>
+    <select asp-for="Version" asp-items="Html.GetEnumSelectList<LuceneVersion>()" class="form-control"></select>
+    <span class="hint">@T["A lucene version to use which can be left with default value."]</span>
+</fieldset>
+
+<fieldset class="form-group" asp-validation-class-for="Analyzer">
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Views/LuceneAnalyzerSettings.Edit.cshtml]: Is there a way to list the available ones? Even if we need to add a new options for that each module can register them. This way we could imagine to provide a new module that would add more analyzers, or special language modules could provide analyzers for these languages.

File: src/OrchardCore.Modules/OrchardCore.Lucene/Views/LuceneAnalyzerSettings.Edit.cshtml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+@using Lucene.Net.Util
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Views/LuceneAnalyzerSettings.Edit.cshtml]: Why isn't this file in the existing LuceneSettings.Edit.cshtml ?

**Comment by sebastienros** [comment]
Since your last change now I don't see any benefit for this PR.

Before we could select a different analyzer from the settings which was an improvement. I suggested that we should be able to do that for each individual index. 

Now we can't change the default analyzer from the UI anymore, and instead you are changing the way analyzers are registered using a provider, which prevents from allowing the selection from the UI.

Right now I am closing this PR. I hope you'll re-open a new one that implements the following requirements: https://github.com/OrchardCMS/OrchardCore/issues/2717

You were very close from that, I hope you will consider this solution.

**Pull request closed without merging**